File: push/src/main/java/top/trumeet/mipushframework/wizard/UsageStatsPermissionInfo.java
package top.trumeet.mipushframework.wizard;

import android.content.Context;
import android.content.Intent;
import android.os.Build;

import androidx.annotation.NonNull;

import com.xiaomi.xmsf.R;

public class UsageStatsPermissionInfo implements PermissionInfo {
    Context context;

    public UsageStatsPermissionInfo(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public PermissionOperator getPermissionOperator() {
        return new UsageStatsPermissionOperator(context);
    }

    @NonNull
    @Override
    public String getPermissionTitle() {
        return context.getString(R.string.wizard_title_stats_permission);
    }

    @NonNull
    @Override
    public String getPermissionDescription() {
        return context.getString(R.string.wizard_title_stats_permission_text);
    }

    @NonNull
    @Override
    public Intent nextPageIntent() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            return PermissionInfoFactory.bindPermissionInfo(
                    new Intent(context, RequestPermissionActivity.class),
                    AlertWindowPermissionInfo.class);
        } else {
            return new Intent(context, FinishWizardActivity.class);
        }
    }

}

File: app/src/main/java/io/github/magisk317/mipush/feature/wizard/permission/UsageStatsPermissionInfo.kt
package io.github.magisk317.mipush.feature.wizard.permission

import android.content.Context
import io.github.magisk317.mipush.runtime.R

class UsageStatsPermissionInfo(private val context: Context) : PermissionInfo {
    override val permissionOperator: PermissionOperator
        get() = UsageStatsPermissionOperator(context)

    override val permissionTitle: String
        get() = context.getString(R.string.wizard_title_stats_permission)

    override val permissionDescription: String
        get() = context.getString(R.string.wizard_title_stats_permission_text)
}
